src/components/Sidebar/ConnectNodeButton.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, memo, useCallback, ChangeEvent, FC } from 'react';
+import { SidebarButton } from 'components/Sidebar';
+import DropDown from 'components/elements/DropDown';
+import useStyles from './styles';
+import { addTransition } from 'utils/builder';
+import { usePathwaysContext } from 'components/PathwaysProvider';
+import { getConnectableNodes } from 'utils/nodeUtils';
+import { useCurrentPathwayContext } from 'components/CurrentPathwayProvider';
+import { useCurrentNodeContext } from 'components/CurrentNodeProvider';
+
+import { faLevelDownAlt } from '@fortawesome/free-solid-svg-icons';
+import { Button } from '@material-ui/core';
+import Tooltip from '@material-ui/core/Tooltip';
+
+const ConnectNodeButton: FC = () => {
+  const { pathway, pathwayRef } = useCurrentPathwayContext();
+  const { currentNode, currentNodeRef } = useCurrentNodeContext();
+  const { updatePathway } = usePathwaysContext();
+
+  const styles = useStyles();
+  const [open, setOpen] = useState(false);
+
+  const options: Array<{ label: string; value: string }> =
+    currentNode && pathway ? getConnectableNodes(pathway, currentNode) : [];
+  const optionsAvailable = options.length > 0;
+
+  const connectToNode = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      const nodeKey = event?.target.value;
+      if (pathwayRef.current && currentNodeRef.current?.key)
+        updatePathway(addTransition(pathwayRef.current, currentNodeRef.current.key, nodeKey));
+      setOpen(false);
+    },
+    [updatePathway, currentNodeRef, pathwayRef]
+  );
+
+  const showDropdown = useCallback(() => {
+    if (optionsAvailable) setOpen(true);
+  }, [optionsAvailable]);
+
+  return (
+    <div>
+      {!open && (
+        <Tooltip
```

Comment:
We want to move this tooltip into the `SidebarButton` component otherwise the hover area is the entire div, not just the button. Moving it also means we don't need to pass in a ref, which is where your ref error is coming from.

To do so, remove the `Tooltip` wrapper here and replace the `SidebarButton` with:

```
<SidebarButton
  buttonName="Connect Node"
  buttonIcon={faLevelDownAlt}
  buttonText="Create a transition to an existing node in the pathway."
  onClick={showDropdown}
  disabled={!optionsAvailable}
  hasTooltip={!optionsAvailable}
  tooltipTitle="There are no possible nodes to connect to."
/>
```
We also want this button to be disabled, so we pass that prop in as well.

Then, we can refactor the `SidebarButton` component as follows:

```
import React, { FC, memo } from 'react';
import { Button, Tooltip } from '@material-ui/core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';

import useStyles from './styles';

interface SidebarButtonProps {
  buttonName: string;
  buttonIcon: IconDefinition;
  buttonText: string;
  extraMargin?: boolean;
  onClick?: () => void;
  hasTooltip?: boolean;
  tooltipTitle?: string;
  disabled?: boolean;
}

const SidebarButton: FC<SidebarButtonProps> = ({
  buttonName,
  buttonIcon,
  buttonText,
  extraMargin = false,
  onClick,
  hasTooltip = false,
  tooltipTitle = '',
  disabled = false
}) => {
  const styles = useStyles();

  const ConditionalTooltip: FC = ({ children }) =>
    hasTooltip ? (
      <Tooltip title={tooltipTitle} placement="top">
        <span>{children}</span>
      </Tooltip>
    ) : (
      <>{children}</>
    );

  return (
    <div className={extraMargin ? styles.sidebarButtonGroupExtraMargin : styles.sidebarButtonGroup}>
      <ConditionalTooltip>
        <Button
          className={styles.sidebarButton}
          variant="contained"
          color="primary"
          startIcon={<FontAwesomeIcon icon={buttonIcon} />}
          onClick={onClick}
          disabled={disabled}
        >
          {buttonName}
        </Button>
      </ConditionalTooltip>
      <div className={styles.sidebarButtonText}>{buttonText}</div>
    </div>
  );
};

export default memo(SidebarButton);
```

Here we use the `hasTooltip` flag to conditionally show the tooltip just on the button. This also allows us to add tooltips in the future on any sidebar button we like.

The last thing we'd need to do is update the disabled styles in the `styles/theme.tsx` file on line 76 as follows:

```
const materialUiOverridesBase = {
  MuiButton: {
    root: {
      borderRadius: 0,
      '&$disabled': {
        backgroundColor: colors.grayMedium,
        color: colors.white
      }
    },
    contained: {
      '&$disabled': {
        backgroundColor: colors.grayMedium,
        color: colors.white
      }
    },
    ...
```